app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/standard/impl/AppEngineStandardCommunityWebIntegration.java

```diff
@@ -49,10 +52,27 @@
   @Override
   public VirtualFile suggestParentDirectoryForAppEngineWebXml(
       @NotNull Module module, @NotNull ModifiableRootModel rootModel) {
-    final VirtualFile root = ArrayUtil.getFirstElement(rootModel.getContentRoots());
+    BuildSystemAppEngineWebXmlDirectoryProvider[] appEngineWebXmlDirectoryProviders =
+        Extensions.getExtensions(BuildSystemAppEngineWebXmlDirectoryProvider.EP_NAME);
+    for (BuildSystemAppEngineWebXmlDirectoryProvider provider : appEngineWebXmlDirectoryProviders) {
+      Optional<String> pathOptional = provider.getAppEngineWebXmlDirectoryPath(module);
+
+      if (pathOptional.isPresent()) {
+        try {
+          return VfsUtil.createDirectoryIfMissing(pathOptional.get());
+        } catch (IOException ioe) {
+          LOG.warn(
```

Comment:
Is this visible enough for a user - i.e. can we suggest some other remedy - like suggest adding simple xml into web resources location?

Reply:
Oh I just realized you have much better messages and notification for missing xml file now in other PRs, thanks!